### DISCUSSION TITLE
@ObservableState in UIKit

### DISCUSSION THREAD

**Discussion by nmalevich** [pointfreeco/swift-composable-architecture#2871] [General]
I have explored migrating guide and projects from examples, but I haven't found answer for my question about observe{} and ObservableState.

For what we can separate observe{} if all of our separated parts of observe execute when we change only one property in ObservableState

**Comment by acosmicflamingo** [comment]
I believe in their videos about Observability, they say it's okay for certain actions to be executed multiple times (like setting a button's attributes to the same value), which is why it should be okay to just utilize one observe closure in your code.

The exception is for computationally-intensive code like reloading a UITableView's data, in which case you do want to have that in a separate closure.

**Comment by mbrandonw** [comment]
Hi @nmalevich, this is discussed in the docs [here](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/objectivec/nsobject/observe(_:)):

> Generally speaking you can usually have a single [observe(_:)](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/objectivec/nsobject/observe(_:)) in the entry point of your view, such as viewDidLoad for UIViewController.
> 
> ...
>
> However, if there is heavy work you need to perform when state changes, then it is best to put that in its own [observe(_:)](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/objectivec/nsobject/observe(_:)).
